include/up-transport-zenoh-cpp/ZenohUTransport.h

```diff
@@ -0,0 +1,102 @@
+// SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache License Version 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef UP_TRANSPORT_ZENOH_CPP_ZENOHUTRANSPORT_H
+#define UP_TRANSPORT_ZENOH_CPP_ZENOHUTRANSPORT_H
+
+#include <up-cpp/transport/UTransport.h>
+
+#include <filesystem>
+#include <optional>
+
+namespace uprotocol::transport {
+
+/// @brief Zenoh implementation of UTransport
+///
+/// This implementation must meet the following requirements:
+///
+/// * [MUST] An implementation MUST support multiple simultaneous instantiatons
+///          from within the same application.
+///
+/// * [MAY]  An implimentation MAY require that each new instance within an
+///          application have a unique configuration from existing instances.
+///
+/// * [MUST] An implementation MUST allow for multiple users of the same
+///          instance.
+///
+/// * [MUST] An implementation MUST be thread-safe.
+///
+/// * [MUST] Throw an exception if the transport fails to initialize or the
+///          configuration is invalid.
+struct ZenohUTransport : public UTransport {
+	/// @brief Constructor
+	///
+	/// @param defaultUri Default Authority and Entity (as a UUri) for
+	///                   clients using this transport instance.
+	/// @param configFile Path to a configuration file containing the Zenoh
+	///                   transport configuration.
+	ZenohUTransport(const v1::UUri& defaultUri,
+	                const std::filesystem::path& configFile);
+
+	virtual ~ZenohUTransport() = default;
+
+protected:
+	/// @brief Send a message.
+	///
+	/// @param message UMessage to be sent.
+	///
+	/// @returns * OKSTATUS if the payload has been successfully
+	///            sent (ACK'ed)
+	///          * FAILSTATUS with the appropriate failure otherwise.
+	[[nodiscard]] virtual v1::UStatus sendImpl(
```

Comment:
Ok I see what you're doing here. You will put the common code in the `send()` implementation and then ask the implementers to do the specific in `sendImpl()`, smart.